From 1ccfd9b8f46012d523413a4e3943281c539cc65e Mon Sep 17 00:00:00 2001
From: Huang Rui <rui1.huang@intel.com>
Date: Mon, 3 Apr 2023 22:49:39 +0800
Subject: [ENABLE] vulkan:fix buf map failure in 32bits app

and when has_tiling_uapi is false
1) NPG: return I915_TILING_NONE which means ISL_TILING_LINEAR
2) On dg2-20221012, it returns -1, will cause vulkan app always crash
3) has_tiling_uapi is always false, because DRM_IOCTL_I915_GEM_SET_TILING
   is not supported in has_get_tiling() function
Signed-off-by: Marc Mao <marc.mao@intel.com>
---
 src/intel/vulkan/anv_gem.c              | 2 +-
 src/intel/vulkan/i915/anv_kmd_backend.c | 6 +++++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/intel/vulkan/anv_gem.c b/src/intel/vulkan/anv_gem.c
index df416cf5674..929fcc1d558 100644
--- a/src/intel/vulkan/anv_gem.c
+++ b/src/intel/vulkan/anv_gem.c
@@ -129,7 +129,7 @@ int
 anv_gem_get_tiling(struct anv_device *device, uint32_t gem_handle)
 {
    if (!device->info->has_tiling_uapi)
-      return -1;
+      return I915_TILING_NONE;
 
    struct drm_i915_gem_get_tiling get_tiling = {
       .handle = gem_handle,
diff --git a/src/intel/vulkan/i915/anv_kmd_backend.c b/src/intel/vulkan/i915/anv_kmd_backend.c
index 562d5b64883..6b9b480b9f2 100644
--- a/src/intel/vulkan/i915/anv_kmd_backend.c
+++ b/src/intel/vulkan/i915/anv_kmd_backend.c
@@ -145,9 +145,13 @@ i915_gem_mmap_offset(struct anv_device *device, struct anv_bo *bo,
    };
    if (intel_ioctl(device->fd, DRM_IOCTL_I915_GEM_MMAP_OFFSET, &gem_mmap))
       return MAP_FAILED;
-
+#ifdef __x86_64__
    return mmap(NULL, size, PROT_READ | PROT_WRITE, MAP_SHARED,
                device->fd, gem_mmap.offset);
+#else
+   return mmap64(NULL, size, PROT_READ | PROT_WRITE, MAP_SHARED,
+               device->fd, gem_mmap.offset);
+#endif
 }
 
 static void *
-- 
2.25.1

